src/main/scala/io/citrine/lolo/trees/ModelNode.scala

```diff
@@ -101,27 +104,16 @@ case class InternalModelNode[+T](
       featureWeights: Map[Int, FeatureWeightFactor]
   ): DenseMatrix[Double] = {
     val featureIndex = split.index
-
+    val goLeft = split.turnLeft(input)
     // The hot node is the one that is selected when the feature is present
-    val (hot, cold) = if (this.split.turnLeft(input)) {
-      (left, right)
-    } else {
-      (right, left)
-    }
-
     // If the feature is omitted or unknown, then the weight assigned to each child is taken
     // as the share of the training data that got split into each child
-    val hotPortion = hot.trainingWeight / trainingWeight
-    val coldPortion = cold.trainingWeight / trainingWeight
+    val (hot, cold) = if (goLeft) (left, right) else (right, left)
+    val (hotPortion, coldPortion) = if (goLeft) (leftPortion, rightPortion) else (rightPortion, leftPortion)
```

Comment:
condensing the previous code which figured out the hot and cold paths then computed the Portions. Now we compute portions once and assign when necessary.